### ISSUE TITLE
getTBSCertificate is broken

### ISSUE THREAD

**Issue by themighty1** [bitpay/bitcore-payment-protocol#15]
it relies on finding zero bytes in the tail. But what is the signature itself contains zero bytes?
I'm afraid there is no way around proper asn1 parsing.

**Comment by themighty1** [comment]
Actually, it is still possible to get away with quick-and-dirty parsing if you do
getTBSCertificate(data, c.signature.data.length);
and then search the tail starting with
data.length - 1 - siglen

**Comment by chjj** [comment]
> I'm afraid there is no way around proper asn1 parsing.

Trust me, I know that much. I spent a long time tearing my damn hair out over asn.1/x.509.

asn1.js was originally going to be used for this, but it was impossible to grab the raw TBS cert out of a signed cert with asn1.js. I was thinking about ripping out of a bunch of code from the asn1.js parser just to get TBS certs, but I tried a simpler solution instead, which worked in practice.

You're right I did overlook the possibility of the sig containing a nul byte. Passing in the sig length we get from the asn1.js output is probably the easiest solution.
